src/publishCommon.ml

```diff
@@ -44,7 +44,12 @@ let repo_of_dir ?(remote="origin") dir =
           eos;
         ])
   in
-  git_query ~dir ["config"; "--get"; "remote."^remote^".url"] >>=
+  (* Use --get-all instead of --get to handle the setup that pushes commits
+     to multiple remote repositories but only fetches commits from the main
+     GitHub repository. For this kind of setups, the first remote repo should
+     be the main repo; using --get will return the last remote repo, which
+     is not we are looking for. *)
+  git_query ~dir ["config"; "--get-all"; "remote."^remote^".url"] >>=
```

Comment:
Shouldn't then take a given line ?

Reply:
The utility function will return the first line. Maybe I misunderstood your question?

Reply:
If it returns the first line, it's ok!

Reply:
Just to be clear, Line 28 drops everything except the first line.
https://github.com/ocaml-opam/opam-publish/blob/0fa83a258dccd857cff6d61889b18d83281811b9/src/publishCommon.ml#L22-L30